online-shop/src/components/Details.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { RouteComponentProps} from 'react-router-dom';
+import products from './productArray';
+import '../App.css';
+import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
+import Grid from '@material-ui/core/Grid';
+import AddShoppingCartIcon from '@material-ui/icons/AddShoppingCart';
+import Button from '@material-ui/core/Button';
+
+type TParams = { id: string };
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    margin: {
+       margin: theme.spacing(1),
+    },
+    button: {
+        margin: theme.spacing(5),
+        fontSize: 25
+    },
+   }),
+);
+
+function Details({match}: RouteComponentProps<TParams>) {
+    const productID = match.params.id;
+    const selectedCategory = products.find(item => item.id === productID);
+    const productName = selectedCategory?.name;
+    const productPrice = selectedCategory?.price;
+    const productCategory = selectedCategory?.category;
+    const productDetails = selectedCategory?.details;
+    const classes = useStyles();
+   
+    return (
+        <div>
+            <div className='SameLine'>
+            <h1 className="Title2"><b>Product:&nbsp;&nbsp;{productName}</b></h1>
+            <Button variant="contained" color="secondary" className={classes.button} startIcon={<AddShoppingCartIcon/>}>
+                Add to cart
+            </Button>
+        </div>
```

Comment:
Add an extra tab space here, to align it with the opening <div> tag from line 35, otherwise it's very confusing to see which closing tag closes which opening tag